src/api.proposed.d.ts

```diff
@@ -97,28 +97,25 @@ declare module './api' {
         /**
          * Returns the list of servers.
          */
-        provideJupyterServers(token: CancellationToken): Promise<JupyterServer[]>;
+        provideJupyterServers(token: CancellationToken): ProviderResult<JupyterServer[]>;
```

Comment:
I notice a lot of `doSomething || []` throughout the diffs so I think a lot of these return types could also have `| undefined`

Reply:
Yes that’s the signature of ProviderResult
It could be a promise, it could be non undefined or defined